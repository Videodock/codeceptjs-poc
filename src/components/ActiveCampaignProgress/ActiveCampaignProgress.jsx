import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import { Typography } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import styles from './ActiveCampaignProgress.styles';

const useStyles = makeStyles(styles);

export default function ActiveCampaignProgress () {
  const { t }   = useTranslation('campaign');
  const classes = useStyles();

  const renderCheckItem = (amountDone, amountTotal, description) => (
    <Grid className={classes.checkItem} xs={12} sm={6} md item>
      <Typography className={classes.amountDone} variant="h3">
        {amountDone}
      </Typography>
      <Typography className={classes.amountTotal} variant="h3">
        /{amountTotal}
      </Typography>
      <Typography variant="body1">
        {description}
      </Typography>
    </Grid>
  );

  return (
    <section className={classes.section}>
      <div className={classes.container}>
        <Grid
          direction="row"
          justify="space-between"
          alignItems="center"
          container
        >
          {renderCheckItem('12k', '345k', t('campaignProgress.clefs_checked'))}
          {renderCheckItem('0', '345k', t('campaignProgress.tempo_checked'))}
          {renderCheckItem('0', '345k', t('campaignProgress.measure_checked'))}
          {renderCheckItem('0', '345k', t('campaignProgress.key_checked'))}
          {renderCheckItem('0', '345k', t('campaignProgress.mei_changes'))}
        </Grid>
      </div>
    </section>
  );
}
