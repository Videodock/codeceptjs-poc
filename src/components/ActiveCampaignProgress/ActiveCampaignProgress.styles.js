import { createStyles } from '@material-ui/styles';

export default ({ palette, typography, spacing, breakpoints }) => createStyles({
  section: {
    boxShadow: '0 1px 6px 0px rgba(0,0,0,0.1)',
  },
  container: {
    paddingTop              : 48,
    paddingBottom           : 48,
    width                   : '80%',
    margin                  : '0 auto',
    [breakpoints.only('xl')]: {
      paddingTop   : 96,
      paddingBottom: 96,
    },
  },
  checkItem: {
    width    : 200,
    height   : 125,
    textAlign: 'center',
  },
  amountDone: {
    color        : palette.primary.main,
    fontSize     : typography.pxToRem(65),
    fontWeight   : 900,
    marginBottom : spacing(-1),
    textTransform: 'uppercase',
  },
  amountTotal: {
    fontSize     : typography.pxToRem(28),
    fontWeight   : 600,
    textTransform: 'uppercase',
  },
});
