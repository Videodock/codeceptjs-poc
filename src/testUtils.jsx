import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
