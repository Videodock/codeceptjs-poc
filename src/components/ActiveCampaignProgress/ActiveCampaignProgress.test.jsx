import React from 'react';
import { render } from '../../testUtils';
import ActiveCampaignProgress from './ActiveCampaignProgress';

describe('<ActiveCampaignProgress />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ActiveCampaignProgress />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
