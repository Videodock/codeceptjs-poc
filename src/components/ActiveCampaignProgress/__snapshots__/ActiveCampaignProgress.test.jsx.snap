// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveCampaignProgress /> renders and matches snapshot 1`] = `
<section
  class="makeStyles-root-1"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      class="MuiGrid-root makeStyles-checkItem-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
    >
      <h3
        class="MuiTypography-root makeStyles-amountDone-3 MuiTypography-h3"
      >
        12k
      </h3>
      <h3
        class="MuiTypography-root makeStyles-amountTotal-4 MuiTypography-h3"
      >
        /
        345k
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Clef's checked
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-checkItem-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
    >
      <h3
        class="MuiTypography-root makeStyles-amountDone-3 MuiTypography-h3"
      >
        0
      </h3>
      <h3
        class="MuiTypography-root makeStyles-amountTotal-4 MuiTypography-h3"
      >
        /
        345k
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Tempo Signature checked
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-checkItem-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
    >
      <h3
        class="MuiTypography-root makeStyles-amountDone-3 MuiTypography-h3"
      >
        0
      </h3>
      <h3
        class="MuiTypography-root makeStyles-amountTotal-4 MuiTypography-h3"
      >
        /
        345k
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Erroneous measure checked
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-checkItem-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
    >
      <h3
        class="MuiTypography-root makeStyles-amountDone-3 MuiTypography-h3"
      >
        0
      </h3>
      <h3
        class="MuiTypography-root makeStyles-amountTotal-4 MuiTypography-h3"
      >
        /
        345k
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Key Signature checked
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-checkItem-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
    >
      <h3
        class="MuiTypography-root makeStyles-amountDone-3 MuiTypography-h3"
      >
        0
      </h3>
      <h3
        class="MuiTypography-root makeStyles-amountTotal-4 MuiTypography-h3"
      >
        /
        345k
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        MEI Editor changes
      </p>
    </div>
  </div>
</section>
`;
